src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -376,7 +376,7 @@ public static void Analyze(CSharpCompilation compilation, MethodSymbol member, B
                 walker.Free();
             }
 
-            if (compilation.LanguageVersion >= MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion())
+            if (compilation.LanguageVersion >= MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion() && compilation.RunNullableWalker)
             {
                 NullableWalker.Analyze(compilation, member, node, diagnostics);
```

Comment:
>Analyze [](start = 31, length = 7)

There are some entry points to NullableWalker (in semantic model, such as ` RewriteNullableBoundNodesWithSnapshots` method) which didn't get the extra check.
I believe this is intentional (you should be able to QuickInfo and see the null-state of a local even in a method inside nullable disable context). Consider adding tests verifying that behavior.

Reply:
> you should be able to QuickInfo and see the null-state of a local even in a method inside nullable disable context

The behavior is intentional, but this is not the reason. If nullability is disabled, it's disabled. We'll get the declared info from any API declarations (if you hover an annotated API you'll see that info, for example), but you won't be able to see local state. The reason they haven't gotten this check is because they're using `EnableNullableSemanticAnalysis` to condition instead. There are already tests that show this in `NullablePublicAPITests.cs`

---
In reply to: [296934289](https://github.com/dotnet/roslyn/pull/36714#discussion_r296934289) [](ancestors = 296934289)